src/Resources/contao/library/Contao/Widget.php

```diff
@@ -1484,20 +1525,12 @@ public function getEmptyStringOrNull()
 	 */
 	public static function getEmptyStringOrNullByFieldType($sql)
 	{
-		if ($sql == '')
+		if (empty($sql))
 		{
 			return '';
 		}
 
-		// Strip the field type definition
-		list(, $def) = explode(' ', $sql, 2);
-
-		if (strpos($def, 'NULL') !== false && strpos($def, 'NOT NULL') === false)
-		{
-			return null;
-		}
-
-		return '';
+		return static::getEmptyValueByFieldType($sql) === null ? null : '';
```

Comment:
This change is wrong. The method comment says:

> Return either an empty string or null based on the SQL string

Therefore we cannot use `getEmptyValueByFieldType()` here!